core/embed/rust/src/ui/model_tr/component/flow.rs

```diff
@@ -148,12 +150,16 @@ where
     fn event_consumed_by_current_choice(&mut self, ctx: &mut EventCtx, pos: ButtonPos) -> bool {
         if matches!(pos, ButtonPos::Left) && self.current_page.has_prev_page() {
             self.current_page.go_to_prev_page();
-            self.scrollbar.inner_mut().go_to_previous_page();
+            self.scrollbar
+                .inner_mut()
+                .set_active_page(self.page_counter as usize, ctx);
```

Comment:
I think this needs to be reverted, because in case there is "more pages withing a single page", the `self.page_counter` in this higher component is always the same, but the page has changed within `self.current_page` - see the screenshots in overall comment

Reply:
Also a possibility, I agree with this.

I little preferred the scrollbar to have the option to paginate on its own - as there would be no need to calculate anything from outside, just tell it "go_to_previous_page" without any arguments